rlp/rlp.go

```diff
@@ -0,0 +1,43 @@
+package rlp
+
+import (
+	"encoding/binary"
+	"math"
+)
+
+type Item struct {
+	D []byte
+	L []*Item
+}
+
+func Encode(input *Item) []byte {
+	if input.D != nil {
+		if len(input.D) == 1 && input.D[0] < 128 {
+			return input.D
+		}
+		return append(encodeLength(input.D, 128), input.D...)
+	}
+
+	var out []byte
+	for i := range input.L {
+		out = append(out, Encode(input.L[i])...)
+	}
+	return append(encodeLength(out, 192), out...)
+}
+
+func encodeLength(input []byte, offset uint8) []byte {
+	switch l := uint64(len(input)); {
```

Comment:
Just double checking here: uint64 is large enough to hold this length, correct? Since the binary representation of the length can't be larger than 191-183=8 bytes long?

Reply:
Honestly, I don't like the current level of type conversions in this code. That being said, this type conversion exists to appease the compiler when it checks how we compare `l` with other values.

Recall the signature for len: `func len(v Type) int`. It seems safe to convert this int to a uint64.

Also, it's worth pointing out that a uint64 can count up to 18 million TB.